utils.py

```diff
@@ -17,6 +21,27 @@ def perform_notebook_conversion(notebook_json):
     return nb_html
 
 
+def perform_rmd_conversion(stream):
+    incoming_data = FileStorage(stream)
+    
+    # The rmarkdown converter unfortunately only works with files.
+    # So we create temp files for the source markdown and destination html data.
+    # The temp files are deleted as soon as the below with block ends.
+    with NamedTemporaryFile(suffix=".Rmd") as in_file:
+        incoming_data.save(dst=in_file.name)
+
+        # Call R rmarkdown package from python.
+        # See https://cran.r-project.org/web/packages/rmarkdown/index.html
+        rmd = importr("rmarkdown")
+        rmd.render(in_file.name)
+
+        out_path = os.path.splitext(in_file.name)[0] + ".html"
+        out_file = open(out_path)
+        res = out_file.read()
+        os.remove(out_path)
```

Comment:
Not enclosing the highlighted code in a `with … as …` block may cause `out_file` to not be closed or deleted in case of exceptions. Could use `try-finally` if `with … as …` too cumbersome here.

Reply:
Fixed, thanks